Golem/Golem.cs

```diff
@@ -230,6 +249,15 @@ public async Task Stop()
             if (!IsRunning() && Status != GolemStatus.Error)
                 return;
 
+            await DetachSuspendHandler();
+
+            await InternalStop();
+        }
+
+        private async Task InternalStop() {
+            if (!IsRunning() && Status != GolemStatus.Error)
+                return;
```

Comment:
I'm a bit unsure about the logic of the early returns and the suspend handler - it looks a bit like the program could exhibit some surprising behaviors.
For example, if I read the code correctly, suspend and resume now restarts Golem if it errored before the process.
It is more or less  a resource - shouldn't we use `try ... finally` for releasing it? Or are we afraid that if some exception is thrown by the function's logic we may be unable to deregister it?

Reply:
Nvm, I hallucinated detaching the handler in `InternalStop()`.
The logic is simple, there's no need to complicate it with `try ... finally`.